contracts/MarginPool.sol

```diff
@@ -77,7 +77,7 @@ contract MarginPool is Ownable {
         uint256 _amount
     ) public onlyController {
         require(_amount > 0, "MarginPool: transferToPool amount is equal to 0");
-
+        require(_user != address(this), "MarginPool: cannot transfer assets from oneself");
```

Comment:
This check is not needed as the `safeTransferFrom` will fail if `_user == address(this)`.